packages/components/src/List/ListComposition/Manager/hooks/useCollectionSelection.hook.js

```diff
@@ -7,6 +7,20 @@ export default function useCollectionSelection(
 ) {
 	const [selectedIds, setSelectedIds] = useState(initialSelectedIds);
 
+	useEffect(() => {
+		// Filter selected items to only preserve the ones that exist in the collection
+		if (selectedIds.length === 0) {
+			return;
+		}
+
+		const availableIds = collection.map(item => item[idKey]);
+		const filteredSelection = selectedIds.filter(id => availableIds.includes(id));
+
+		if (filteredSelection.length !== selectedIds.length) {
```

Comment:
You can remove this `if`. If the collection or the ids change, we want to always impact the state.